File: InMotionGIT.Common - Copy/DataType/LookUpValueExtend.vb
Imports System.ComponentModel
Imports System.Runtime.Serialization
Imports System.Xml.Serialization

Namespace DataType

    <DataContract(Namespace:="urn:InMotionGIT.Common.DataType")>
    <Serializable()>
    <XmlType(Namespace:="urn:InMotionGIT.Common.DataType")>
    <XmlRoot(Namespace:="urn:InMotionGIT.Common.DataType")>
    <Attributes.TypeStructure("Code")>
    Public Class LookUpValueExtend
        Inherits LookUpValue

        <DataMember()>
        <XmlAttribute(), DefaultValue("")>
        Public Property ShortDescription As String

        <DataMember()>
        <XmlAttribute(), DefaultValue("")>
        Public Property ParentCode As String

        Public Sub New()
            MyBase.New()
        End Sub

        Public Sub New(ByVal code As String)
            Me.Code = code
        End Sub

        Public Sub New(ByVal code As String, ByVal description As String)
            Me.Code = code
            Me.Description = description
        End Sub

        Public Sub New(ByVal code As String, ByVal description As String, shortDescription As String)
            Me.Code = code
            Me.Description = description
            Me.ShortDescription = shortDescription
        End Sub

        Public Sub New(ByVal code As String, ByVal description As String, shortDescription As String, parentCode As String)
            Me.Code = code
            Me.Description = description
            Me.ShortDescription = shortDescription
            Me.ParentCode = parentCode
        End Sub

    End Class

End Namespace

File: Common/InMotionGIT.Common/DataType/LookUpValueExtend.cs
using System;
using System.ComponentModel;
using System.Runtime.Serialization;
using System.Xml.Serialization;

namespace InMotionGIT.Common.DataType
{

    [DataContract(Namespace = "urn:InMotionGIT.Common.DataType")]
    [Serializable()]
    [XmlType(Namespace = "urn:InMotionGIT.Common.DataType")]
    [XmlRoot(Namespace = "urn:InMotionGIT.Common.DataType")]
    [Attributes.TypeStructure("Code")]
    public class LookUpValueExtend : LookUpValue
    {

        [DataMember()]
        [XmlAttribute()]
        [DefaultValue("")]
        public string ShortDescription { get; set; }

        [DataMember()]
        [XmlAttribute()]
        [DefaultValue("")]
        public string ParentCode { get; set; }

        public LookUpValueExtend() : base()
        {
        }

        public LookUpValueExtend(string code)
        {
            Code = code;
        }

        public LookUpValueExtend(string code, string description)
        {
            Code = code;
            Description = description;
        }

        public LookUpValueExtend(string code, string description, string shortDescription)
        {
            Code = code;
            Description = description;
            ShortDescription = shortDescription;
        }

        public LookUpValueExtend(string code, string description, string shortDescription, string parentCode)
        {
            Code = code;
            Description = description;
            ShortDescription = shortDescription;
            ParentCode = parentCode;
        }

    }

}